nightfall-deployer/README.md

```diff
@@ -0,0 +1,180 @@
+    `amount` is the determined max allowed value for deposits and withdrawals. One entry for `MATIC` is required. The placeholder `NETWORK_NAME` should be 
+    replaced with the network name:
+      ```
+        tokens: {
+            [NETWORK_NAME]: [
+            {
+              name: 'MATIC',
+              address: '0x0000000000000000000000000000000000001010',
+              amount: '1000000000000000000000',
+            },
+            ...
+          ],
+          ...
+        }
+      ```
+    Follows a sample:
+      ```
+        tokens: {
+            mumbai: [
+            {
+              name: 'WETH',
+              address: '0xa6fa4fb5f76172d178d61b04b0ecd319c5d1c0aa',
+              amount: '1000000000000000000',
+            },
+            {
+              name: 'MATIC',
+              address: '0x0000000000000000000000000000000000001010',
+              amount: '1000000000000000000000',
+            },
+            {
+              name: 'USDC',
+              address: '0xe6b8a5CF854791412c1f6EFC7CAf629f5Df1c747',
+              amount: '1000000000',
+            },
+            {
+              name: 'stMATIC',
+              address: '0xa337f0B897a874DE1E9F75944629a03F911cFbE8',
+              amount: '1000000000',
+            },
+            ...
+          ],
+          ...
+        }
+      ```
+
+4. Start the deployment with the following command, where the parameter is the .env file created in the `Step 1.`.
+
+    ```
+    ./bin/deploy-contracts [.env.deployment.file]
+    ```
+    e.g.
+    ```
+    ./bin/deploy-contracts .env.deployment.my_network
+    ```
+
+    During the deployment an instance of the `worker` application is required to be up, so that the Circuits can be generated accordingly. This instance is started up 
+    automatically, but it is possible configure the worker server if one is already available, for this set `CIRCOM_WORKER_HOST` in the .env file.
+
+    There is a variable in the .env file called `DEPLOYMENT_SERVICES`. It allows one to set the services to start up. This variable can be passed over with the command
+    as well.
+    ```
+    DEPLOYMENT_SERVICES=client,optimist,worker ./bin/deploy-contracts .env.deployment.my_network
+    ```
+
+One should remember to have enough funds when performing the deployment so that it can finish properly.
+
+After the deployment finishs successfully, one can find the files that were generated during the deployment under the folder `docker/volumes`:
+- `build/contracts`: contain the ABI files.
+- `proving_files`: contain circuits' files.
+
+These files are going to be used by the applications `nightfall-client` & `nightfall-optimist`.
+
+## Testing
+For testing purposes, some addresses under `TEST_OPTIONS.addresses` in `config/defaults.js` are funded with `MATIC`. Set the variable `DEPLOY_MOCK_TOKENS` in the .env file to `false` to prevent this behavior (This will also prevent the deployment of the `ERC20Mock` Smart Contract).
```

Comment:
Actually, I think there's a Mock contract per ERC (?). So you could either make a general reference "Mock ERC contracts", or just list the 3 "ERC20Mock, ERC721Mock and ERC1155Mock".

```suggestion
For testing purposes, some Ethereum addresses are funded with `MATIC` - see `TEST_OPTIONS.addresses` in `config/defaults.js` . Set the variable `DEPLOY_MOCK_TOKENS` in the .env file to `false` to prevent this behaviour. This will also prevent the deployment of the `ERC20Mock` Smart Contract.
```